plugins/auth-node/src/DefaultIdentityClient.ts

```diff
@@ -75,18 +80,27 @@ export class DefaultIdentityClient implements IdentityApi {
     this.algorithms = options.hasOwnProperty('algorithms')
       ? options.algorithms
       : ['ES256'];
+    this.serverTokenManager = options.serverTokenManager;
   }
 
   async getIdentity({ request }: IdentityApiGetIdentityRequest) {
-    if (!request.headers.authorization) {
+    const token = getBearerTokenFromAuthorizationHeader(
+      request.headers.authorization,
+    );
+
+    if (!token) {
       return undefined;
     }
+
     try {
-      return await this.authenticate(
-        getBearerTokenFromAuthorizationHeader(request.headers.authorization),
-      );
+      return await this.authenticate(token);
     } catch (e) {
-      throw new NotAllowedError(e.message);
+      if (this.serverTokenManager) {
+        // really this should return some sore of identity that represents the backend.
+        await this.serverTokenManager.authenticate(token);
```

Comment:
There is a lot of other stuff in this patch, but this bit needs consideration.

@freben @Rugvip